packages/kbn-interpreter/src/public/interpreter.js

```diff
@@ -17,55 +17,46 @@
  * under the License.
  */
 
-import { socketInterpreterProvider } from '../common/interpreter/socket_interpret';
+import { interpreterProvider } from '../common/interpreter/interpreter_provider';
 import { serializeProvider } from '../common/lib/serialize';
 import { createHandlers } from './create_handlers';
 
-export async function initializeInterpreter(socket, typesRegistry, functionsRegistry) {
-  let resolve;
-  const functionList = new Promise(_resolve => (resolve = _resolve));
-
-  const getInitializedFunctions = async () => {
-    return functionList;
-  };
+export async function initializeInterpreter(kfetch, typesRegistry, functionsRegistry) {
```

Comment:
I have no issue with doing it this way, just curious if there's a technical reason for passing kfetch in instead of just importing it. I suspect it has to do with not including it in some bundle, but I'm not sure. Maybe it just makes testing easier, which is great too.

Reply:
Yeah. `ui/*` packages can't be imported into anything that lives within `/packages` so we have to inject.